shared/modules/mv3.utils.js

```diff
@@ -1,11 +1,22 @@
 import browser from 'webextension-polyfill';
 
+const runtimeManifest =
+  global.chrome?.runtime.getManifest() || global.browser?.runtime.getManifest();
+
 /**
- * A boolean indicating whether the manifest of the current extension
- * is set to manifest version 3.
+ * A boolean indicating whether the manifest of the current extension is set to manifest version 3.
+ *
+ * We have found that when this is run early in a service worker process, the runtime manifest is
+ * unavailable. That's why we have a fallback using the ENABLE_MV3 constant. The fallback is also
+ * used in unit tests.
  */
-export const isManifestV3 =
```

Comment:
There were a few changes made here in #26059 that I included in this cherry-pick as well, as my changes depended on them. Specifically:
* This constant is now exported
* The fallback on `process.env.ENABLE_MV3` was added

I did this rather than cherry-picking the entirety of #26059 because that PR impacted a much wider scope, would have been more complex and riskier to back port in its entirety. cc @HowardBraham

Reply:
Whoops, one more change from #26059 needed: https://github.com/MetaMask/metamask-extension/pull/26582/commits/ff880544443cf63af2e0d39699be69db218b8556